pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -56,6 +56,18 @@ func (l *Prometheus) GetStatus() PrometheusStatus {
 	return l.Status
 }
 
+// +kubebuilder:validation:Enum=ResourceName;ShardName
+type AdditionalLabelSelector string
```

Comment:
(nit) we can define the enum values as const and reuse in the controller.
```suggestion
type AdditionalLabelSelector string

const (
    ResourceNameLabelSelector AdditionalLabelSelector = "ResourceName"
    ShardNameLabelSelector AdditionalLabelSelector = "ShardName"
    
)
```